### PR TITLE
new example: sprite animation in response to an event

### PR DESCRIPTION
**Pull request by awwsmm**
# Objective

- animating a sprite in response to an event is a [common beginner problem](https://www.reddit.com/r/bevy/comments/13xx4v7/sprite_animation_in_bevy/)

## Solution

- provide a simple example to show how to animate a sprite in response to an event

### CODE REVIEW

**Comment by awwsmm** [comment]
https://github.com/bevyengine/bevy/assets/29188350/2f5fd2be-9c4f-4582-a927-45c98f251b6b

The left sprite is animated when the user clicks the left arrow key. The right sprite is animated when the user clicks the right arrow key. The sprites are offset horizontally for visibility. The two sprites are animated at different FPSes so the user can compare them.

**Review by alice-i-cecile — APPROVED** [review]
I like this example, and think it should exist. I suspect we'll want to make it more sophisticated in time, but this is an excellent place to start.

File: examples/2d/sprite_animation.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        // press the left arrow key to animate the left sprite
+        .add_systems(
+            Update,
+            trigger_animation::<LeftSprite>.run_if(input_just_pressed(KeyCode::ArrowLeft)),
+        )
+        // press the right arrow key to animate the right sprite
+        .add_systems(
+            Update,
+            trigger_animation::<RightSprite>.run_if(input_just_pressed(KeyCode::ArrowRight)),
+        )
```
Comment (matiqo15) [inline examples/2d/sprite_animation.rs:24]: We should also add a basic ui to describe this keys for the wasm example

**Review by matiqo15 — APPROVED** [review]
Looks good, just a small suggestion

**Review by pablo-lua — APPROVED** [review]
All good. I only have one concern: I think we should probably have more comments in the main systems explaining what we are doing and why we are doing. For example, explaining that `execute_animations` will basically tick between all the sprites in the texture atlas and return to first after `X` ticks, or explaining what are we spawning in the setup function, probably this will help beginners in the future.

File: examples/2d/sprite_animation.rs (hunk 1 above)
Comment (awwsmm) [reply examples/2d/sprite_animation.rs:24]: Screenshot 2024-04-23 at 12 40 55

Done!

**Comment by awwsmm** [comment]
> All good. I only have one concern: I think we should probably have more comments in the main systems explaining what we are doing and why we are doing. For example, explaining that `execute_animations` will basically tick between all the sprites in the texture atlas and return to first after `X` ticks, or explaining what are we spawning in the setup function, probably this will help beginners in the future.

Done! Added some explanatory comments.

File: examples/2d/sprite_animation.rs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        RightSprite,
+        animation_config_2,
+    ));
+
+    // create a minimal UI explaining how to interact with the example
+    commands.spawn(TextBundle {
+        text: Text::from_section(
+            "Left Arrow Key: Animate Left Sprite\nRight Arrow Key: Animate Right Sprite",
+            TextStyle {
+                font_size: 40.0,
```
Comment (mockersf) [inline examples/2d/sprite_animation.rs:146]:
```suggestion
                font_size: 20.0,
```

**Comment by mockersf** [comment]
added two suggestions to have the same ui style as most other examples

**Comment by awwsmm** [comment]
> added two suggestions to have the same ui style as most other examples

Applied!

**Pull request merged**